samples/embedded-auth-with-sdk/src/main/java/com/okta/spring/example/controllers/HomeController.java

```diff
@@ -176,8 +181,32 @@ public ModelAndView displayForgotPasswordPage(final HttpSession session) {
      */
     @GetMapping("/register")
     public ModelAndView displayRegisterPage(final HttpSession session) {
-        begin(session);
-        return new ModelAndView("register");
+        AuthenticationResponse authenticationResponse = begin(session);
+
+        authenticationResponse =
+                authenticationWrapper.fetchSignUpFormValues(authenticationResponse.getProceedContext());
+
+        ModelAndView modelAndView = new ModelAndView("register");
+
+        Optional<FormValue> userProfileFormValue = authenticationResponse.getFormValues()
+                    .stream()
+                    .filter(x -> x.getName().equals("userProfile"))
+                    .findFirst();
+
+
+        if (!userProfileFormValue.isPresent()) {
+            displayErrorPage();
```

Comment:
Should return the "error" ModelAndView here to prevent execution last piece of the code

```java
return displayErrorPage();
```

Reply:
You mean, I should change it to this? 
```
        if (!userProfileFormValue.isPresent()) {
            return new ModelAndView("error");
        }
```

Reply:
No, to this:

```java
if (!userProfileFormValue.isPresent()) {
    return displayErrorPage();
}
```